### PR TITLE
Refine KLDCriterion

### PR DESCRIPTION
**Pull request by yangw1234**
## What changes were proposed in this pull request?

1. Make the KLDCriterion average over a sub-batch, so that the scale of the loss is independent of batch size.

2. Expose two node apis in python

## How was this patch tested?

unit tests

### CODE REVIEW

**Comment by yangw1234** [comment]
pr validation pass

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/KLDCriterion.scala [hunk 1]
```diff
@@ -39,31 +42,36 @@ class KLDCriterion[@specialized(Float, Double) T: ClassTag](
...
 
+    val batchSize = mean.size(1)
+
     //  Appendix B from VAE paper: -0.5 * sum(1 + log(sigma^2) - mu^2 - sigma^2)
     mean.pow(ev.fromType(2))
     vars.resizeAs(logVar).copy(logVar).exp()
     logVar.add(ev.one).add(ev.fromType(-1), mean).add(ev.fromType(-1), vars)
 
-    output = ev.times(ev.fromType(-0.5), logVar.sum())
+    output = ev.times(ev.fromType(-0.5 / batchSize), logVar.sum())
```
Comment (zhangxiaoli73) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/KLDCriterion.scala]: add a parameter "sizeAverage" to decide whether you need to div batchsize

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/KLDCriterion.scala [hunk 2]
```diff
@@ -27,6 +27,9 @@ import scala.reflect.ClassTag
  * The input has to be a table. The first element of input is the mean of the distribution,
  * the second element of input is the log_variance of the distribution. The input distribution is
  * assumed to be diagonal.
+ *
+ * The mean and log_variance are both assumed to be two dimensional tensors. The first dimension are
```
Comment (zhangxiaoli73) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/KLDCriterion.scala]: update api docs

File: pyspark/bigdl/nn/layer.py [hunk 3]
```diff
@@ -53,6 +53,13 @@ def of(cls, jvalue, bigdl_type="float"):
     def element(self):
         return Layer.of(self.value.element())
 
+    def remove_pre_edges(self):
```
Comment (zhangxiaoli73) [inline pyspark/bigdl/nn/layer.py]: refine KLDCriterion, why do you have to change this?

**Comment by zhangxiaoli73** [comment]
LGTM

**Comment by yangw1234** [comment]
pr validation pass

**Pull request merged**